File: src/com/charter/enterprise/collatz/util/CollatzInfoPrinter.java
package com.charter.enterprise.collatz.util;

import com.charter.enterprise.collatz.model.CollatzDataModel;

import java.util.Collections;
import java.util.List;

public class CollatzInfoPrinter {

    public void prettyPrintCollatzList(List<CollatzDataModel> collatzList) {
        System.out.flush();

        Collections.sort(collatzList);

        for(CollatzDataModel collatzDataModel : collatzList){
            prettyPrintCollatzDataModel(collatzDataModel);
        }
    }

    public void prettyPrintCollatzDataModel(CollatzDataModel collatzDataModel){
        System.out.format("The number %5d has %5d collatz iterations!%n",
                collatzDataModel.getCollatzNumber(),
                collatzDataModel.getCollatzIterations());
    }

}

File: src/util/collatz-info-printer.ts
import { CollatzDataModel } from '../model/collatz-data-model';

export class CollatzInfoPrinter {
    prettyPrintCollatzList(collatzList: CollatzDataModel[]): void {
        collatzList.sort((a, b) => a.compareTo(b));

        for (const collatzDataModel of collatzList) {
            this.prettyPrintCollatzDataModel(collatzDataModel);
        }
    }

    prettyPrintCollatzDataModel(collatzDataModel: CollatzDataModel): void {
        const number = collatzDataModel.getCollatzNumber().toString().padStart(5);
        const iterations = collatzDataModel.getCollatzIterations().toString().padStart(5);
        console.log(`The number ${number} has ${iterations} collatz iterations!`);
    }
}
